test/e2e/features_test.go

```diff
@@ -703,3 +703,95 @@ func TestDefaultIngressClass(t *testing.T) {
 		return false
 	}, ingressWait, time.Second)
 }
+
+// TestMissingCRDsDontCrashTheController ensures that in case of missing CRDs installation in the cluster, specific
+// controllers are disabled, this fact is properly logged, and the controller doesn't crash.
+func TestMissingCRDsDontCrashTheController(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	clusterBuilder := kind.NewBuilder()
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		clusterBuilder.WithClusterVersion(clusterVersion)
+	}
+	cluster, err := clusterBuilder.Build(ctx)
+	require.NoError(t, err)
+	var addons []clusters.Addon
+	addons = append(addons, metallb.New())
+
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+
+	builder := environments.NewBuilder().WithExistingCluster(cluster).WithAddons(addons...)
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+
+	t.Logf("building cleaner to dump diagnostics...")
+	cleaner := clusters.NewCleaner(env.Cluster())
+	defer func() {
+		if t.Failed() {
+			output, err := cleaner.DumpDiagnostics(ctx, t.Name())
+			t.Logf("%s failed, dumped diagnostics to %s", t.Name(), output)
+			assert.NoError(t, err)
```

Comment:
Similar thing as in https://github.com/Kong/gateway-operator/pull/223#discussion_r984712411

Reply:
f3bc3b9